File: ch01/ch01_05.rb
def word_ngram(text, n)
  words = text.split
  (0..words.length-n).map { |i| words[i..i+n-1] }
end

def letter_ngram(text, n)
  (0..text.length-n).map { |i| text[i..i+n-1] }
end

if $0 == __FILE__
  text = 'I am an NLPer'
  p word_ngram(text, 2)
  p letter_ngram(text, 2)
end

File: ch01/ch01_05.py
def word_ngram(text, n):
    words = text.split()
    return [words[i:i+n] for i in range(len(words)-n+1)]

def letter_ngram(text, n):
    return [text[i:i+n] for i in range(len(text)-n+1)]

if __name__ == '__main__':
    text = 'I am an NLPer'
    print(word_ngram(text, 2))
    print(letter_ngram(text, 2))
